src/controllers/hid/hidioreport.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "controllers/controller.h"
+#include "controllers/hid/hiddevice.h"
+#include "util/compatibility/qmutex.h"
+#include "util/duration.h"
+
+class HidIoReport {
+  public:
+    HidIoReport(const unsigned char& reportId, const unsigned int& reportDataSize);
+
+    /// Caches new report data, which will later send by the IO thread
+    void cacheOutputReport(const QByteArray& data,
```

Comment:
The naming of these function is still not intuitive. How about:

- `updateOutputReportData()`
- `sendNextOutputReport()`

Reply:
I renamed the class and it members:
`HidIoReport` to `HidIoOutputReport`
`HidIoReport::cacheOutputReport` to `HidIoOutputReport::updateCachedData`
`HidIoReport::sendOutputReport` to `HidIoOutputReport::sendCachedData`
 I hope the naming is now more intuitive.